### DISCUSSION TITLE
Split repo into library part and ffmpeg part

### DISCUSSION THREAD

**Discussion by lukasf** [ffmpeginteropx/FFmpegInteropX#295] [Ideas]
We are currently maintaining two separate components: Our FFmpegInteropX library, and the FFmpegUWP build. While the lib depends on the FFmpeg build, both are pretty separate, with separate versions and releases.

Sometimes it is a bit confusing to maintain two different libs and have two version numbers in one repo. There is also the downside that people who just want to try the lib and maybe debug a bit, will pull the complete repo with all huge submodules. But none of the submodules is actually required for building or debugging the lib. Sometimes people even think they have to manually build ffmpeg before building and using the lib.

Should we split up the repo and move the FFmpeg build into a separate repo?

**Comment by brabebhin** [comment]
I like the way it is right now but I can also see your point about forcing people to download ffmpeg. So from my perspective, we can split them.

**Comment by brabebhin** [comment]
We should also consider taking the BasicVideoEffect and its related implementations out of the main library and distribute it separately. It introduces the win2D dependency which may not be needed/wanted by everyone and it will make the core library easier to maintain.
